pkg/auth/tokenparser_test.go

```diff
@@ -0,0 +1,111 @@
+package auth_test
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/auth"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	testutils "github.com/codeready-toolchain/registration-service/test"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTokenParser(t *testing.T) {
+	// create logger and registry.
+	logger := log.New(os.Stderr, "", 0)
+	configRegistry := configuration.CreateEmptyRegistry()
+
+	// create test keys
+	tokengenerator := testutils.NewTokenManager()
+	kid0 := uuid.NewV4().String()
+	_, err := tokengenerator.AddPrivateKey(kid0)
+	require.NoError(t, err)
+	kid1 := uuid.NewV4().String()
+	_, err = tokengenerator.AddPrivateKey(kid1)
+	require.NoError(t, err)
+
+	// create two test tokens, both valid
```

Comment:
you could move this part to the only subtest that uses it, to simplify the understanding of all this setup

Reply:
Done d7a998c.